blockchain/blockindex.go

```diff
@@ -225,13 +272,33 @@ func (bi *blockIndex) AddNode(node *blockNode) {
 	bi.Unlock()
 }
 
-// RemoveNode removes the provided node from the block index.  There is no check
-// whether another node in the index depends on this one, so it is up to caller
-// to avoid that situation.
+// GetStatus provides concurrent-safe access to the status field of a node.
+//
+// This function is safe for concurrent access.
+func (bi *blockIndex) GetStatus(node *blockNode) blockStatus {
```

Comment:
Please call this `NodeStatus` instead of `GetStatus`.  Naming functions `GetX` is highly discouraged by the [Effective Go Guidelines](https://golang.org/doc/effective_go.html#Getters) which this project follows.